plugins/inventory/aci.py

```diff
@@ -0,0 +1,166 @@
+        self.params = dict()
+
+        validator = ArgumentSpecValidator(argument_spec)
+        result = validator.validate(parameters)
+
+        if result.error_messages:
+            display.vvv("Validation failed: {0}".format(", ".join(result.error_messages)))
+
+        self.params = result.validated_parameters
+
+    @property
+    def tmpdir(self):
+        # if _ansible_tmpdir was not set and we have a remote_tmp,
+        # the module needs to create it and clean it up once finished.
+        # otherwise we create our own module tmp dir from the system defaults
+        if self._tmpdir is None:
+            basedir = None
+
+            basefile = "ansible-moduletmp-%s-" % time.time()
+            try:
+                tmpdir = tempfile.mkdtemp(prefix=basefile, dir=basedir)
+            except (OSError, IOError) as e:
+                self.fail_json(msg="Failed to create remote module tmp path at dir %s " "with prefix %s: %s" % (basedir, basefile, to_native(e)))
+            atexit.register(shutil.rmtree, tmpdir)
+            self._tmpdir = tmpdir
+
+        return self._tmpdir
+
+    def warn(self, warning):
+        display.vvv(warning)
+
+    def fail_json(self, msg, **kwargs) -> t.NoReturn:
+        raise AnsibleError(msg)
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable):
+
+    NAME = "cisco.aci.aci"
+
+    def verify_file(self, path):
+        """return true/false if this is possibly a valid file for this plugin to consume"""
+        valid = False
+        if super(InventoryModule, self).verify_file(path):
+            # base class verifies that file exists and is readable by current user
+            if path.endswith(("cisco_aci.yaml", "cisco_aci.yml")):
+                valid = True
+        return valid
+
+    def parse(self, inventory, loader, path, cache=True):
+
+        # call base method to ensure properties are available for use with other helper methods
+        super(InventoryModule, self).parse(inventory, loader, path, cache)
+
+        # this method will parse 'common format' inventory sources and
+        # update any options declared in DOCUMENTATION as needed
+        config = self._read_config_data(path)
+        config.update(state="query")
+
+        argument_spec = aci_argument_spec()
+        argument_spec.update(
+            id=dict(type="int", aliases=["controller", "node"]),
+            state=dict(type="str", default="query", choices=["query"]),
+            keyed_groups=dict(type="list"),
+            plugin=dict(type="str"),
+        )
+
+        module = MockAnsibleModule(
+            argument_spec=argument_spec,
+            parameters=config,
+        )
+
+        aci = ACIModule(module)
+        aci.construct_url(root_class=dict(aci_class="topSystem"))
+
+        aci.get_existing()
+
+        # parse data and create inventory objects:
+        for device in aci.existing:
+            attributes = device.get("topSystem", {}).get("attributes")
+            if attributes.get("name"):
```

Comment:
this could still be none and thus lead to an attribute error

Reply:
I don't think so.  

if None will not validate to true and therefore skip adding that host,

Reply:
If attributes is None, how would a get be called on it? I agree that if attributes.get("name") would return None then it would not validate to true and skip.

Reply:
I must not have had enough coffee!  Yes, you are right.  Thank you.